marshalable/src/tests.rs

```diff
@@ -1,3 +1,10 @@
+// TODO potential bug in clippy? we need to have unit values for testing the
+// behavior of marshaling on unit, but the I could not allow this warning for
+// HasUnitField, possibly because of span bug, clippy suggested fix is syntax
+// error possibly due to another bug, and placing the allow attribute on the
+// struct or the let binding does not make the warning go aways
+#![allow(clippy::let_unit_value)]
```

Comment:
@trent-reed one thing to point out is I don't like this, so please fact check that this is actually the best we can do here.

Reply:
Sorry for the delay here, I was thinking too much about this...

Basically, I'm not sure that we even want to support unit values. I think it might be a nicer, cleaner future if we had explicit zero-sized types with actual names that show what they are. But I'm not 100% positive on that either, because I don't know that what we have is the best format for these command types either. In general, I'm still not totally sure on the direction of commands in this library - so that makes this a tad difficult to think about, and the only thing that really uses these unit-types are commands - so... Yeah - hopefully with more changes and effort we can nail down a clearer direction for these things and see then if unit types are still necessary.

I could imagine just always clearly saying there's a type but that it has no values. For example:

```
// May not have any sessions or handles
struct StartupRequest {
  startup_type: TpmSu,
}
struct StartupHandles {}
struct StartupResponse {}
type StartupCommand = TpmCommand<StartupRequest, TpmNoSessions>;
impl TpmCommandCode on StartupCommand {
  // Only doing it this way because we can't use const generics here.
  const CMD_CODE: Tpm2Cc = Tpm2Cc::Startup;
  type Response = StartupResponse;
}

// Example usage:
let command = StartupCommand {
  request: StartupRequest{
    startup_type: TpmSu::Clear,
  },
  handles: StartupHandles{},
  sessions: TpmNoSessions{},
};
let response = client.run(&command)?;
```

But even this isn't a fully-complete idea, and I don't like it for some other reasons... I guess my point is I'm not sure we need unit-type support. But currently we do so I guess the comment is moot as far as this PR is concerned.

---

I will confirm that I also was unable to find a way to make clippy only allow the unit value on a structure for the test, so I'm fine taking this for now but we should see if it remains necessary to use unit types in the future or if named types aren't clearer/better after a `TpmCommand` refactor (which a refactor of commands is kind of what I requested in #63).

Reply:
Actually, you know I just realized what we could have done is limit the impact of this by putting that one test in a sub-module and marking that sub-module as `allow(clippy::let_unit_value)`, to limit the exposure of this. I'll send a PR for that if you agree that'd be a good thing to do.

Reply:
I was just about to  think of a really better solution, just opening an issue in clippy asking them to fix their stuff,  I just need to do minimum reproducer. feel free to separate that one test,  but I feel it might be un-necessary, it is just a clippy warning in tests.

Reply:
Apparently, we can fix this bug on our side by using `quote` instead of `quote_spanned`. But that will reduce quality of diagnostics.